e2e/test/scenarios/admin/performance/preemptiveCaching.cy.spec.ts

```diff
@@ -0,0 +1,133 @@
+const { H } = cy;
+
+import { WRITABLE_DB_ID } from "e2e/support/cypress_data";
+import {
+  createNativeQuestion,
+  createQuestion,
+  getTableId,
+  visitQuestion,
+} from "e2e/support/helpers";
+
+import {
+  advanceServerClockBy,
+  interceptPerformanceRoutes,
+  resetServerTime,
+} from "./helpers/e2e-performance-helpers";
+import {
+  durationRadioButton,
+  enablePreemptiveCaching,
+  openSidebarCacheStrategyForm,
+  saveCacheStrategyForm,
+} from "./helpers/e2e-strategy-form-helpers";
+
+const CACHE_DURATION_MS = 24 * 60 * 60 * 1000;
+
+const WRITABLE_TEST_TABLE = "scoreboard_actions";
+
+describe(
+  "Preemptive caching for MBQL and native queries",
+  { tags: "@external" },
+  () => {
+    describe("ee", () => {
+      beforeEach(() => {
+        resetServerTime();
+        interceptPerformanceRoutes();
+        H.restore("postgres-writable");
+        cy.signInAsAdmin();
+        H.setTokenFeatures("all");
+
+        H.resetTestTable({ type: "postgres", table: WRITABLE_TEST_TABLE });
+        H.resyncDatabase({
+          dbId: WRITABLE_DB_ID,
+          tableName: WRITABLE_TEST_TABLE,
+        });
+      });
+
+      function visitCachedQuestion(questionId: string) {
```

Comment:
Probably we don't need cy.then here, also you could define it within `runPreemptiveCachingTest` to remove the need to provide `questionId` every time
```
      function runPreemptiveCachingTest(questionId: string) {
        function visitCachedQuestion() {
          cy.log("Visiting the question");
          visitQuestion(questionId);
        }

        visitCachedQuestion();
        // ...
      }

```